### PR TITLE
Fed stats privacy2 [skip ci]

### PR DESCRIPTION
**Pull request by chesterxgchen**
1)  Implement statistics privacy filter
2) Remove Fed Stats privacy related validation and logics 
3) updated the unit tests and readme for the new changes

4) handle None filter case in fl_conf.py. When scope did not specify filter, such as empty scope, the component will be None.
5) re-word job_scope to privacy scope in app_deployer.py
6)  job_runner.py swallow the failure details, add it back
7) give more details on which configuration has the typos or mis-configuration when failure in json_configer.py

### CODE REVIEW

**Review by nvidianz — DISMISSED** [review]

File: nvflare/private/fed/utils/app_deployer.py [hunk 1]
```diff
@@ -83,4 +83,5 @@ def deploy(self) -> str:
                 return "not authorized"
 
         except BaseException as ex:
-            return f"exception {ex} when deploying app {self.app_name}"
+            raise Exception(f"exception {ex} when deploying app {self.app_name}")
+            # return f"exception {ex} when deploying app {self.app_name}"
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/utils/app_deployer.py:87]: the app_deployer was originally just returning "string", but this PR change to raise Exception.

will this cause any issues?

File: nvflare/app_common/filters/statistics/statistics_privacy_filter.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+from nvflare.apis.dxo import DXO, DataKind
+from nvflare.apis.dxo_filter import DXOFilter
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+from nvflare.app_common.app_constant import StatisticsConstants as StC
+from nvflare.app_common.filters.statistics.metrics_privacy_cleanser import MetricsPrivacyCleanser
+from nvflare.fuel.utils import fobs
+
+
+class StatisticsPrivacyFilter(DXOFilter):
```
Comment (YuanTingHsieh) [inline nvflare/app_common/filters/statistics/statistics_privacy_filter.py:26]: suggest just put this file inside `app_common/filters`

File: nvflare/app_common/filters/statistics/min_max_noise_generator.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+import random
+from typing import Tuple
+
+from nvflare.apis.fl_component import FLComponent
+from nvflare.app_common.app_constant import StatisticsConstants as StC
+from nvflare.app_common.filters.statistics.metrics_privacy_cleanser import MetricsPrivacyCleanser
+
+
+class AddNoiseToMinMax(FLComponent, MetricsPrivacyCleanser):
```
Comment (YuanTingHsieh) [inline nvflare/app_common/filters/statistics/min_max_noise_generator.py:23]: suggest put these "cleansers" concept inside `app_common/statistics`

File: examples/federated_statistics/local/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+Statistics privacy filters are task result filters.
+```
+StatisticsPrivacyFilter
+```
+The StatisticsPrivacyFilter is consists of three MetricsPrivacyFilters focused on the metrics sent
+from client to server. 
+
+StatisticsPrivacyFilter can be considered as an interceptor before the results delivered to server. 
+Currently, we use three MetricsPrivacyFilters to guard the data privacy
```
Comment (YuanTingHsieh) [inline examples/federated_statistics/local/README.md:36]: MetricsPrivacyFilters or MetricsPrivacyCleansers or MetricsPrivacyCheckers?

**Review by YuanTingHsieh — APPROVED** [review]
Thanks LGTM

**Pull request merged**